Added HTML Inputs Extension

samuellouf opened the pull request:
# HTML Extension
Hi there!
This extension allows you to insert **HTML inputs on the canvas**.
Here are the blocks.
![This extension's blocks](https://github.com/TurboWarp/extensions/assets/104774504/4086e9b7-a24d-4609-9fd4-9e076a4b4225)
You can :
- create/delete inputs and select menus. 
- set/get their values, dimentions, position.
- Hide/show them...

Here's [the extension](https://github.com/samuellouf/extensions/blob/HTML/extensions/SamuelLouf/HTML.js).
Here's the exemple's [.sb3](https://samuellouf.github.io/turbowarp_extensions/HTML-exemple.sb3) / [compiled html](https://samuellouf.github.io/turbowarp_extensions/HTML-form.html).

Enjoy!

samuellouf committed:
Created HTML.js (my extension)

samuellouf committed:
Merge branch 'TurboWarp:master' into HTML

samuellouf committed:
Update HTML.js

Made my code prettier.

mybearworld commented:
Isn't the position functionality already in the iframe extension?

samuellouf commented:
> Isn't the position functionality already in the iframe extension?

Yes, it is. But, let me explain a little bit :
In the iframe extension, the Scratch VM will create a div over the canvas to add the iframe. But, when you insert HTML, it will just create a data URL for the code and set it as the iframe's source. 
**Example:**
`<h1>It works!</h1>` will become `<iframe src="data:text/html;,%3Ch1%3EIt%20works!%3C%2Fh1%3E"></iframe>`
You can't get an input's value and change the elements' coordinates individually.
With my extension, you can.
In my extension, the Scratch VM will create a div over the canvas to add the element. It will do this for every single element.

nmsderp commented:
> > Isn't the position functionality already in the iframe extension?
> 
> Yes, it is. But, let me explain a little bit : In the iframe extension, the Scratch VM will create a div over the canvas to add the iframe. But, when you insert HTML, it will just create a data URL for the code and set it as the iframe's source. **Example:** `<h1>It works!</h1>` will become `<iframe src="data:text/html;,%3Ch1%3EIt%20works!%3C%2Fh1%3E"></iframe>` You can't get an input's value and change the elements' coordinates individually. With my extension, you can. In my extension, the Scratch VM will create a div over the canvas to add the element. It will do this for every single element.

seems interesting, should probably add a simplified version of what you said to the description of the extension so people get the difference

CST1229 commented:
Yea, considering what this extension does, maybe it should be called something like "HTML Inputs" or something.

samuellouf committed:
Changed a few things

Update and rename HTML.js to HTMLInputs.js
Turned insertHTML into HTMLInputs

samuellouf commented:
> Yea, considering what this extension does, maybe it should be called something like "HTML Inputs" or something.

Done!

yuri-kiss commented:
Closed for being stale.

samuellouf commented:
I'm going to work on it! (Yep, I know, I'm 11 months late)

samuellouf committed:
Merge branch 'TurboWarp:master' into HTML

yuri-kiss commented:
okay, I will reopen the PR

samuellouf commented:
Forgot to add `Scratch.translate`

samuellouf committed:
Fixed lint, removed useless spaces and line skips

samuellouf committed:
Okay now I formatted it

samuellouf committed:
Merge branch 'TurboWarp:master' into HTML

samuellouf commented:
Okay! My pull request is validated!

samuellouf committed:
Merge branch 'TurboWarp:master' into HTML

samuellouf committed:
Made a few fixes

 - Translated menus
 - Added code to remove the elements when the project is stopped
 - Optimized "getElement"

samuellouf committed:
Passed the checks

samuellouf committed:
Merge branch 'TurboWarp:master' into HTML

samuellouf commented:
Finally done!

samuellouf commented on extensions/SamuelLouf/HTMLInputs.js:
```diff
@@ -0,0 +1,676 @@
+// ID: samuelloufhtmlinputs
+// Description: Add HTML inputs over the stage.
+// By: SamuelLouf <https://scratch.mit.edu/users/samuellouf/>
+
```
Also done

samuellouf commented on extensions/SamuelLouf/HTMLInputs.js:
```diff
@@ -0,0 +1,676 @@
+      };
+    }
+
+    createInput({ TYPE, NAME }) {
```
Did that

samuellouf reviewed (commented):
Done

samuellouf committed:
Merge branch 'TurboWarp:master' into HTML

samuellouf committed:
Merge branch 'TurboWarp:master' into HTML

samuellouf committed:
Merge branch 'TurboWarp:master' into HTML

samuellouf committed:
Changed the URL, License, added JSDoc

samuellouf committed:
Merge branch 'HTML' of https://github.com/samuellouf/extensions into HTML

samuellouf committed:
Format

samuellouf committed:
Changed URL, fixed banner
